packages/core/src/util/properties-util.ts

```diff
@@ -17,10 +17,8 @@ export const nonEnumerable = (target: any, propertyKey: string) => {
  * @returns A boolean denoting whether an entity is a navigation property or not.
  */
 export function isNavigationProperty<EntityT extends EntityBase>(
-  key: keyof EntityT,
-  entityConstructor: Constructable<EntityT>
+  key: string,
+  entityConstructor: any
```

Comment:
I assume the weakening of the type was unavoidable?

Reply:
It was impractical from the get-go